arch/risc-v/src/common/supervisor/riscv_perform_syscall.c

```diff
@@ -46,6 +46,16 @@ void *riscv_perform_syscall(uintreg_t *regs)
     {
       (*running_task)->xcp.regs = regs;
     }
+  else
+    {
+#ifdef CONFIG_ARCH_ADDRENV
+      addrenv_switch(NULL);
```

Comment:
let's create a new patch which try the best to pass tcb addrenv_switch to avoid query the current task again in addrenv_switch

Reply:
ok

Reply:
https://github.com/apache/nuttx/pull/15035